x/bank/types/send_authorization.go

```diff
@@ -47,3 +79,15 @@ func (a SendAuthorization) ValidateBasic() error {
 	}
 	return nil
 }
+
+func validateAllowedAddresses(allowed []sdk.AccAddress) ([]string, error) {
+	if len(allowed) == 0 {
+		return nil, nil
+	}
+
+	allowedAddrs := make([]string, len(allowed))
```

Comment:
Can we also add validation for duplicate addresses?

Reply:
added in `func (a SendAuthorization) ValidateBasic()`